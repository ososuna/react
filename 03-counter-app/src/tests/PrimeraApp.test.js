
import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import PrimeraApp from "../PrimeraApp";

describe('<PrimeraApp /> tests', () => {
    
    // test('should show the message "Hola Mundo"', () => {
        
    //     const saludo = 'Hola Mundo';
        
    //     const { getByText } = render( <PrimeraApp saludo='Hola Mundo' /> );

    //     expect( getByText( saludo ) ).toBeInTheDocument();

    // })

    test('should show <PrimeraApp /> correctly', () => {
        
        const saludo = 'Hola Mundo';
        const wrapper = shallow( <PrimeraApp saludo = { saludo } /> );
        
        expect( wrapper ).toMatchSnapshot();

    });
    

    test('should show the props subtitle', () => {
        
        const saludo = 'Hola Mundo';
        const subtitle = 'This is a subtitle';
        
        const wrapper = shallow(
            <PrimeraApp
                saludo = { saludo }
                subtitulo = { subtitle }
            />
        );
        
        const textParagraph = wrapper.find('p').text();
        
        expect( textParagraph ).toBe( subtitle );

    });

})

