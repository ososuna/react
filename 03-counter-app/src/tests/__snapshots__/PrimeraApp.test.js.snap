// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimeraApp /> tests should show <PrimeraApp /> correctly 1`] = `
<Fragment>
  <h1>
    Hola Mundo
  </h1>
  <p>
    Subt√≠tulo
  </p>
</Fragment>
`;
